From 89991c49b26e6881baee9aa836336eae40e533ff Mon Sep 17 00:00:00 2001
From: Jizhou Liao <Jizhou.Liao@nxp.com>
Date: Thu, 9 Nov 2017 16:31:00 -0800
Subject: [PATCH 1/4] Fix GKI task release twice issue

GKI task "BTU_TASK" is released twice during NFC de-initialization,
it causes memory corruption.

Test: check log during NFC de-initialization

Change-Id: Ia78b55466ba8a7369fd47722c2be7352149cfe42
---
 src/gki/ulinux/gki_ulinux.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gki/ulinux/gki_ulinux.c b/src/gki/ulinux/gki_ulinux.c
index cb02add..70eb1d0 100644
--- a/src/gki/ulinux/gki_ulinux.c
+++ b/src/gki/ulinux/gki_ulinux.c
@@ -1110,6 +1110,9 @@ uint8_t GKI_resume_task(uint8_t task_id) {
 **
 *******************************************************************************/
 void GKI_exit_task(uint8_t task_id) {
+  if (task_id >= GKI_MAX_TASKS) {
+    return;
+  }
   GKI_disable();
   gki_cb.com.OSRdyTbl[task_id] = TASK_DEAD;
 
-- 
2.17.0

